src/engine/Resources/Sound.ts

```diff
@@ -582,6 +607,25 @@ class WebAudioInstance implements IAudio {
 
       return this._playingPromise;
    }
+   
+   public playWithProfile(){
+     if (this._isPaused) {
+       this._resume();
+     } else if (!this._isPlaying) {
+       this._start();
+     }
+     var point1 = 0.8;
+     var point2 = 0.1;
+     var point3 = 0.1;
+     var point4 = 0.4;
+     
+     while (new Date().getTime() < this._startTime + this._buffer.duration*1000) {
+       var cpt = (new Date().getTime() - this._startTime) / (this._buffer.duration*1000) ;
```

Comment:
The more I think about it, it seems `currentTime` may be useful to track current playthrough time in a song, although `this._startTime` may not be used then, which I suppose is no problem